### PR TITLE
docs: add CCloud KSQL migration guide content (DOCS-3728)

### PR DESCRIPTION
**Pull request by JimGalasyn**

### CODE REVIEW

**Comment by derekjn** [comment]
@JimGalasyn the last piece of this is to reference the migration guide from the CSU docs. Can you add the below section right before the "Sizing guidelines" section on this page?

https://docs.confluent.io/current/cloud/features.html#pricing

---

**Scaling CSUs**

Once your application has been provisioned, CSUs may only be changed with a support request. A CSU scaling request will be processed with whatever priority you assign it. If you do not have a support plan, you will need to provision a new application with the desired number of CSUs. If you use this approach for scaling, you may want to [migrate your old application](link to migration guide) to your new one.

**Comment by JimGalasyn** [comment]
@derekjn, I tweaked the intro, let me know what you think. I'll add the link from the CSU docs in another PR, since that topic lives in the main docs repo.

**Comment by derekjn** [comment]
Looks great @JimGalasyn! Only thing I'd suggest changing is that PA means "production available" so we should probably use that in place of "publicly available".

**Comment by JimGalasyn** [comment]
kk cool, how's this look?

File: docs/installation/cloud-ksql-migration-guide.rst [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+Migrate |ccloud| KSQL from the preview version to the production-available version
+##################################################################################
+
+To deliver the benefits of new features and improvements as soon as they become
+available, new versions of |ccloud| KSQL are deployed frequently. Most of these
+deployments are invisible to you as a user, requiring no action on your part.
+
+The production-available version of |ccloud| KSQL is not backward-compatible 
+with the preview version, but after the release of the production-available
+version, we're committed to preserving backward compatibility.
```
Comment (vcrfxia) [inline docs/installation/cloud-ksql-migration-guide.rst]: @derekjn is this strictly true? I thought we had plans for supporting multiple versions in the future, in order to account for breaking changes if necessary.

File: docs/installation/cloud-ksql-migration-guide.rst [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+.. note::
+
+    We will never automatically upgrade your KSQL application to use a
+    backward-incompatible release.
+
+Port the application’s schema
+*****************************
+
+You can run the following commands by using the |ccloud| web interface or the
+KSQL CLI. If you choose to use the KSQL CLI, you may find the SPOOL command
```
Comment (vcrfxia) [inline docs/installation/cloud-ksql-migration-guide.rst]: Should we link to the docs for using the KSQL CLI with CCloud KSQL here? We've seen a bunch of questions on how to do that.

File: docs/installation/cloud-ksql-migration-guide.rst (hunk 1 above)
Comment (derekjn) [reply docs/installation/cloud-ksql-migration-guide.rst]: Yes, I only added this part after @rodesai reviewed the initial draft (which didn't make this strong of a commitment) and suggested this.

File: docs/installation/cloud-ksql-migration-guide.rst (hunk 2 above)
Comment (derekjn) [reply docs/installation/cloud-ksql-migration-guide.rst]: Probably a good idea, here's the link we should target when referencing the KSQL CLI:

https://docs.confluent.io/current/quickstart/cloud-quickstart/ksql.html#accessing-a-ksql-application-in-ccloud-with-api-keys

**Pull request merged**